$MirOS: ports/emulators/qemu/patches/patch-hw_pc_c,v 1.2 2005/11/15 12:25:09 tg Exp $
--- hw/pc.c.orig	Sun Nov  6 18:20:37 2005
+++ hw/pc.c	Thu Nov 17 14:30:38 2005
@@ -416,7 +416,7 @@ static void pc_init1(int ram_size, int v
     char buf[1024];
     int ret, linux_boot, initrd_size, i, nb_nics1;
     unsigned long bios_offset, vga_bios_offset;
-    int bios_size, isa_bios_size;
+    int bios_size, isa_bios_size, vga_bios_size;
     PCIBus *pci_bus;
 
     linux_boot = (kernel_filename != NULL);
@@ -448,8 +448,12 @@ static void pc_init1(int ram_size, int v
     } else {
         snprintf(buf, sizeof(buf), "%s/%s", bios_dir, VGABIOS_FILENAME);
     }
+    vga_bios_size = get_image_size(buf);
+    if (vga_bios_size != 32768)
+	    goto bios_error;
+
     ret = load_image(buf, phys_ram_base + vga_bios_offset);
-    
+
     /* setup basic memory access */
     cpu_register_physical_memory(0xc0000, 0x10000, 
                                  vga_bios_offset | IO_MEM_ROM);
